src/EnergyPlus/LowTempRadiantSystem.cc

```diff
@@ -1972,34 +1971,53 @@ namespace LowTempRadiantSystem {
 
         if (state.dataLowTempRadSys->FirstTimeInit) {
 
-            state.dataLowTempRadSys->ZeroSourceSumHATsurf.dimension(state.dataGlobal->NumOfZones, 0.0);
-            state.dataLowTempRadSys->QRadSysSrcAvg.dimension(state.dataSurface->TotSurfaces, 0.0);
-            state.dataLowTempRadSys->LastQRadSysSrc.dimension(state.dataSurface->TotSurfaces, 0.0);
-            state.dataLowTempRadSys->LastSysTimeElapsed.dimension(state.dataSurface->TotSurfaces, 0.0);
-            state.dataLowTempRadSys->LastTimeStepSys.dimension(state.dataSurface->TotSurfaces, 0.0);
+            for (RadNum = 1; RadNum <= state.dataLowTempRadSys->NumOfHydrLowTempRadSys; ++RadNum) {
+                auto &thisLTR = state.dataLowTempRadSys->HydrRadSys(RadNum);
+                thisLTR.QRadSysSrcAvg.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastQRadSysSrc.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastSysTimeElapsed.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastTimeStepSys.dimension(thisLTR.NumOfSurfaces, 0.0);
+            }
+            for (RadNum = 1; RadNum <= state.dataLowTempRadSys->NumOfCFloLowTempRadSys; ++RadNum) {
+                auto &thisLTR = state.dataLowTempRadSys->CFloRadSys(RadNum);
+                thisLTR.QRadSysSrcAvg.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastQRadSysSrc.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastSysTimeElapsed.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastTimeStepSys.dimension(thisLTR.NumOfSurfaces, 0.0);
+            }
+            for (RadNum = 1; RadNum <= state.dataLowTempRadSys->NumOfElecLowTempRadSys; ++RadNum) {
+                auto &thisLTR = state.dataLowTempRadSys->ElecRadSys(RadNum);
+                thisLTR.QRadSysSrcAvg.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastQRadSysSrc.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastSysTimeElapsed.dimension(thisLTR.NumOfSurfaces, 0.0);
+                thisLTR.LastTimeStepSys.dimension(thisLTR.NumOfSurfaces, 0.0);
+            }
```

Comment:
Why are these variables in the LowTempRadiantSystem allocated to the number of surfaces while the HIghTempRadiantSystem holds a scalar?

Reply:
Note here that LastSysTimeElapsed and LastTimeStepSys are arrays which all elements would hold the same value. These could be scalars like all the others (better think through that logic before changing it)? I think I also saw this in VentilatedSlabs so check there as well.

Reply:
Hmmm...yeah, I'll have to think about that a little more.  I am pretty sure that the time quantities do not need to be arrays.  Those are going to be set at the system level, not at the surface level.  I'll go back and look at the LTR and Ventilated Slab and think that through some more and then make changes as necessary.  Hopefully I can look at this tomorrow some more after I finish up another defect for which I am in the midst of putting a unit test together.  Thanks again for your input on this!

Reply:
3 object loops in a row.

Reply:
-3